packages/cli-hydrogen/src/cli/services/build/check-lockfile.ts

```diff
@@ -0,0 +1,94 @@
+import {file, git, path} from '@shopify/cli-kit'
+import {renderWarning} from '@shopify/cli-kit/node/ui'
+import {lockfiles} from '@shopify/cli-kit/node/node-package-manager'
+import type {Lockfile} from '@shopify/cli-kit/node/node-package-manager'
+
+function missingLockfileWarning() {
+  renderWarning({
+    headline: 'No lockfile found',
+    body:
+      `If you don’t commit a lockfile, then your app might install the wrong ` +
+      `package versions when deploying. To avoid versioning issues, generate a ` +
+      `new lockfile and commit it to your repository.`,
+    nextSteps: [
+      [
+        'Generate a lockfile. Run',
+        {
+          command: 'npm|yarn|pnpm install',
+        },
+      ],
+      'Commit the new file to your repository',
+    ],
+  })
+}
+
+function tooManyLockfilesWarning(lockfiles: Lockfile[]) {
+  const packageManagers = {
+    'yarn.lock': 'yarn',
+    'package-lock.json': 'npm',
+    'pnpm-lock.yaml': 'pnpm',
+  }
+
+  const lockfileList = lockfiles.map((lockfile) => {
+    return `• ${lockfile} (created by ${packageManagers[lockfile]})`
```

Comment:
This makes me think that we need a new token type, maybe a `list` so you could specify something like:

```
body: [
   'some text',
   {
      list: {
        items: ['1', '2', '3']
      }
    }
]
```

I'm going to put a reminder to look back at this work once this token is introduced and update this code.

Reply:
cc @MeredithCastile @Arkham